lib/Db/AliasMapper.php

```diff
@@ -47,4 +49,22 @@ public function findAll($accountId, $currentUserId) {
 		];
 		return $this->findEntities($sql, $params);
 	}
+
+	/**
+	 * @param int $accountId the account whose aliases will be deleted
+	 * @param string $currentUserId the user that is currently logged in
+	 */
+	public function deleteAll($accountId, $currentUserId) {
+		$sql = 'DELETE FROM *PREFIX*mail_aliases USING *PREFIX*mail_accounts
```

Comment:
- [ ] Blocked: switch db mappers to query builder to make sure queries like these actually work across all supported dbs. This has to happen in another PR. I will open a ticket for that.

Reply:
-> https://github.com/nextcloud/mail/issues/960

Reply:
I've finally had some time to work on this. See https://github.com/nextcloud/mail/pull/1036.